### ISSUE TITLE
Floating Panel dismiss button

### ISSUE THREAD

**Issue by rolson** [Esri/arcgis-maps-sdk-swift-toolkit#923]
I'm not quite sure where the problem is here. If I add a close button to the floating panel with a utility network trace component, I get this which looks good some of the time:

<img width="300" alt="image" src="https://github.com/user-attachments/assets/7a89de2e-0d7c-4d77-ad57-57db6156f923">

But in some cases when the floating panel is collapsed, the close button is hidden:

<img width="300" alt="image" src="https://github.com/user-attachments/assets/7494bfea-5fd6-4412-af9f-a66801482dc3">

Is there a better way to support the workflow where you want to dismiss the floating panel and cancel the trace?

**Comment by mhdostal** [comment]
The example does not allow closing the trace, but I can see the benefit in having a "Cancel" button, maybe alongside the "Trace" and "Clear All Results" buttons. 

The "Clear All Results" button gets cut off in your second screen shot; the "Trace" button, prior to running a trace, does not; there must be a slight difference in how those two are handled.

**Comment by rolson** [comment]
It's not necessarily "cancel". But more like "Done" or "Close". And I don't know that it belongs in the trace component. I think it's the app developer to decide when and how to close it on devices like a phone where it's covering a lot of the real estate. I think this is more of a floating panel issue potentially. Or something odd in the way the floating panel and the trace component work together.

**Comment by dfeinzimer** [comment]
Coincidentally I received a question this morning from Kotlin about support for programatic trace starting points which got me thinking about a potential workaround for this in interim lieu of custom header support or a close button on the Floating Panel.

Replace `UtilityNetworkTraceExampleView.swift` with the attached version to try it.

[UtilityNetworkTraceExampleView.swift.zip](https://github.com/user-attachments/files/17496696/UtilityNetworkTraceExampleView.swift.zip)


https://github.com/user-attachments/assets/c39397e6-a905-4bea-bf1e-bc5c4edb5be3

**Comment by rolson** [comment]
I like that idea. In practice, this doesn't work that well. Maybe it's just a bug in the example code and could be fixed. Also, tapping "Add New" raises questions about what it should do in that case.

**Comment by dfeinzimer** [comment]
> Also, tapping "Add New" raises questions about what it should do in that case.

Agreed, opened issue https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/issues/925

**Comment by rolson** [comment]
I can't get the UNTrace component to work well in a floating panel when trying to add content (ie Title and Dismiss button) above it. These things just aren't quite playing nicely together. Is there a way to limit the floating panel detent to full and half?

**Comment by dfeinzimer** [comment]
> Is there a way to limit the floating panel detent to full and half? That could work for my scenario.

Not directly currently. I'd think monitoring (and programmatically setting when desired) the bound `selectedDetent` should work for that though.

**Comment by rolson** [comment]
I wonder if it would be good to have a built-in way to dismiss the floating panel. Since the drag bar is on the panel, maybe there should be some sort of option for a dismiss/close button. Also the panel has a binding to the isPresented, so it could easily set that to false.

**Comment by dfeinzimer** [comment]
> I wonder if it would be good to have a built-in way to dismiss the floating panel.

I think this definitely makes sense to add.

**Comment by mhdostal** [comment]
Some additional info:

When a trace takes more than a second or two to complete, it's impossible to know if the trace is running. It would be good to have an activity indicator to give user feedback that the trace is in progress.

TK "Utility Network Trace" component

Trace Setup:
The map is zoomed in a little bit from the initial extent

![Image](https://github.com/user-attachments/assets/e15c5144-3cd8-4d9b-99f7-18a1d2d48d31)

Results:

![Image](https://github.com/user-attachments/assets/111b78a0-59a9-4966-8394-a0cf73785b84)

TK v200.7
iPad Pro 11"
iPadOS 18.3.2
